src/main/resources/avro/bdg.avdl

```diff
@@ -988,6 +703,97 @@ record VariantAnnotation {
 }
 
 /**
+ Allele encodings for genotypes.
+ */
+enum Allele {
+
+  /**
+   Reference allele.
+   */
+  REF,
+
+  /**
+   Alternate allele.
+   */
+  ALT,
+
+  /**
+   An unspecified other alternate allele. This may occur when the variant for
+   the genotype partially encoded by this allele is split as a multi-allelic site.
+   */
+  OTHER_ALT,
+
+  /**
+   Missing allele.
+   */
+  NO_CALL
+}
+
+/**
+ Genotype.
+ */
+record Genotype {
+
+  /**
+   The variant for this genotype.
+   */
+  union { null, Variant } variant = null;
+
+  /**
```

Comment:
Yes, nested fields are still a big problem, mostly in the case when we are trying to minimize latency (Mango). One of the motivations for putting in `contigName/start/end` fields was because of this latency issue.